### PR TITLE
release 1.32 sql extension rtm hotfix

### PR DESCRIPTION
**Pull request by brian-harris**
Public preview SQL Migration bug fixes
* encode rest api calls to support database names with Unicode characters.
* refactor getMigationStatus error detection and meta data collection and retention logic
* removed secrets from migration status before save and after load
* add null checks to handle undefined properties in incomplete migration status responses
* adds autocutover configuration
* adds empty table status
* fix content display logic

### CODE REVIEW

File: extensions/sql-migration/src/api/azure.ts [hunk 1]
```diff
@@ -124,7 +124,7 @@ export type SqlVMServer = {
 };
 export async function getAvailableSqlVMs(account: azdata.Account, subscription: Subscription, resourceGroup: azureResource.AzureResourceResourceGroup): Promise<SqlVMServer[]> {
 	const api = await getAzureCoreAPI();
-	const path = `/subscriptions/${subscription.id}/resourceGroups/${resourceGroup.name}/providers/Microsoft.SqlVirtualMachine/sqlVirtualMachines?api-version=2017-03-01-preview`;
+	const path = encodeURI(`/subscriptions/${subscription.id}/resourceGroups/${resourceGroup.name}/providers/Microsoft.SqlVirtualMachine/sqlVirtualMachines?api-version=2017-03-01-preview`);
```
Comment (Charles-Gagnon) [inline extensions/sql-migration/src/api/azure.ts:127]: @aasimkhan30 We should look into doing this in azurecore instead - this isn't something specific to sql-migration unless I'm missing something. Doing it for all Azure REST calls seems reasonable.

**Comment by Charles-Gagnon** [comment]
Are you planning on adding the extension back into the release at some point then?

**Comment by brian-harris** [comment]
> Are you planning on adding the extension back into the release at some point then?

yes in coordination with @aasimkhan30 we pulled the extension allow more time to test these changes along side a service update.  The next steps are to validate these changes along side a service update, then coordinate the delayed addition of the extension into the gallery.

File: extensions/sql-migration/src/api/azure.ts (hunk 1 above)
Comment (aasimkhan30) [reply extensions/sql-migration/src/api/azure.ts:127]: I agree. I will add that to azurecore. However, we still need this piece of code in the extension as it won't be available before next release.

**Review by kburtram — CHANGES_REQUESTED** [review]
I think the plan will to ship directly from a branch forked off of the release/1.32 rather than merge into the release branch.  Let's sync on Teams to chat about any logistics if necessary.  Thanks!

**Comment by Charles-Gagnon** [comment]
@brian-harris Are you planning on getting this merged in at some point?

**Comment by rchlkm** [comment]
I believe we can close this PR. I believe the plan is to ship directly without merging into release branch.
We have also updated the main branch with any changes made on this branch.

**Comment by kburtram** [comment]
Closing this since I think the extension is being built from its own branch.  Please reactive if this still needs to be ported.  Thanks!

**Pull request closed without merging**